numba/unicode.py

```diff
@@ -869,6 +872,13 @@ def unicode_strip_types_check(chars):
         raise TypingError('The arg must be a UnicodeType or None')
 
 
+def args_types_check(arg):
+    if not (arg is None or isinstance(arg, (types.Omitted,
+                                            types.Optional,
```

Comment:
The type of the `Optional` should be checked to ensure it also is an integer type, for example:
```python
from numba import njit

@njit
def foo(x, y):
    if len(x) > 14:
        maybe_none = 'oops'
    else:
        maybe_none = None
    return x.count(y, maybe_none, 1)

a = "ascii"
b = "a"

print(foo.py_func(a, b))
print(foo(a, b))
```
fails strangely. One way to do this is something like `if isinstance(arg, types.Optional) and not isinstance(arg.type, types.Integer: raise ...`. A test should be added to ensure this is correctly implemented.